### PR TITLE
Use nginx as frontend

### PR DESCRIPTION
**Pull request by jmolmo**
First changes to start using Nginx as Ansible Runner Service API frontend

Just tested that i can build the container and the API is accesible. This means that uwsgi and nginx are running. :-)

Signed-off-by: Juan Miguel Olmo Martínez <jolmomar@redhat.com>

### CODE REVIEW

File: misc/nginx/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## Local Preparation
+1. download the project to your home directory and untar/unzip
+2. create an archive of the project called ansible-runner-service.tar.gz and
+store it in the misc/docker directory
+3. set up you local environment to persist the config
+```
+ sudo mkdir /etc/ansible-runner-service
+ sudo mkdir -p /usr/share/ansible-runner-service/{artifacts,env,inventory, project}
```
Comment (Daniel-Pivonka) [inline misc/nginx/README.md]: space between ", project}" cause me issues creating the directories

Comment (jmolmo) [reply misc/nginx/README.md]: Thank you for testing it Daniel, i will remove the extra space.!!

The aim of this first review was only to present the pieces and how i'm configuring them. In this way i can have a quick feedback and change things (even components) if something is not ok.

In next PR i will add the SSL support with the mutual TLS authentication that will solve the problem you see about the ssl version number.

**Comment by jmolmo** [comment]
Now Mutual TLS authentication is working with all the current API endpoints!!

Still user identification & token utilization needed... in next PR all this stuff will be removed.

**Comment by Daniel-Pivonka** [comment]
so far everything has worked for me.

**Comment by jmolmo** [comment]
Tests ok:
https://travis-ci.com/jmolmo/ansible-runner-service/builds/107073578

File: misc/docker/Dockerfile [hunk 2]
```diff
@@ -18,8 +18,8 @@ RUN yum -y install epel-release  && \
 RUN easy_install-3.6 -d /usr/lib/python3.6/site-packages pip && \
     ln -s /usr/lib/python3.6/site-packages/pip3 /usr/local/bin/pip3
 RUN /usr/local/bin/pip3 install crypto docutils psutil paramiko PyYAML \
-                 pyOpenSSL flask flask-restful pyjwt && \
-    /usr/local/bin/pip3 install --no-cache-dir ansible-runner && \
+                 pyOpenSSL flask flask-restful && \
+    /usr/local/bin/pip3 install --no-cache-dir ansible-runner==1.3.0 && \
```
Comment (matburt) [inline misc/docker/Dockerfile]: 1.3.2 was just released!

Comment (jmolmo) [reply misc/docker/Dockerfile]: Updated!

**Comment by jmolmo** [comment]
Unit tests passed:
 https://travis-ci.com/jmolmo/ansible-runner-service/builds/108097811

And some manual tests with the Ansible Orchestrator in Ceph worked perfectly without any problem.

**Comment by jmolmo** [comment]
Test OK
https://travis-ci.com/jmolmo/ansible-runner-service/builds/108126359

**Comment by jmolmo** [comment]
@matburt , i have included a new Dockerfile using as base the "ansible runner 1.3.2" container image. However this image still is using Python 2.7.  So no possible for us to use Python 3.6 in Ansible Runner Service ( because ansible runner is  installed as a Python 2.7 package).  For the moment  i configre the service to use Python 2.7 ... but i think that we should move to the new Python version asap.

Is planned the update of the Ansible runner 1.3.x to use Python 3.6?

**Comment by matburt** [comment]
That's a pretty good point, I *should* start producing py3 and py27 base images. I'll look at doing that over the next day.

**Pull request merged**